src/frontend/src/optimizer/plan_node/stream_dynamic_filter.rs

```diff
@@ -33,18 +36,41 @@ pub struct StreamDynamicFilter {
     pub base: PlanBase<Stream>,
     core: generic::DynamicFilter<PlanRef>,
     cleaned_by_watermark: bool,
+    condition_always_relax: bool,
 }
 
 impl StreamDynamicFilter {
     pub fn new(core: DynamicFilter<PlanRef>) -> Self {
         let watermark_columns = core.watermark_columns(core.right().watermark_columns()[0]);
 
-        // TODO: derive from input
+        // TODO(st1page): the condition is wrong, introduce monotonically increasing property of the node
+        // TODO(st1page): https://github.com/risingwavelabs/risingwave/pull/13984
+        let right_monotonically_increasing = {
+            if let Some(e) = core.right().as_stream_exchange() && *e.distribution() == Distribution::Broadcast {
+                if let Some(proj) = e.input().as_stream_project() {
+                    proj.input().as_stream_now().is_some()
+                } else {
+                    false
+                }
+            } else {
+                false
+            }
+        };
```

Comment:
hack for the temporal filter, but will fix it with monotonic property later. The #13984 is too big and I'd like to merge this PR at first because we only use now executor as the temporal filter and the hack is correct.